providers/linux/procnet.go

```diff
@@ -0,0 +1,112 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package linux
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/go-sysinfo/types"
+)
+
+// fillStruct is some reflection work that can dynamically fill one of our tagged `netstat` structs with netstat data
+func fillStruct(str interface{}, data map[string]map[string]int64) {
+	val := reflect.ValueOf(str).Elem()
+	typ := reflect.TypeOf(str).Elem()
+
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		if tag := field.Tag.Get("netstat"); tag != "" {
+			if values, ok := data[tag]; ok {
+				val.Field(i).Set(reflect.ValueOf(values))
+			}
+		}
+	}
+}
+
+// parseEntry parses two lines from the net files, the first line being keys, the second being values
+func parseEntry(line1, line2 string) (map[string]int64, error) {
+	keyArr := strings.Split(strings.TrimSpace(line1), " ")
+	valueArr := strings.Split(strings.TrimSpace(line2), " ")
+
+	if len(keyArr) != len(valueArr) {
+		return nil, errors.New("Key and value lines are mismatched")
```

Comment:
Error messages should not begin with capitals. https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
There are a few of these in other places.